docs/syntax-table-destructuring.md

```diff
@@ -0,0 +1,137 @@
+# Table Destructuring Syntax
+
+
+
+## Summary
+Introduce new syntax for unpacking values from tables.
+
+```lua
+local &{ foo, ["@bar"] = bar } = thing
+-- foo == thing.foo
+-- bar == thing["@bar"]
+```
+
+This RFC addresses issues brought up from previous table destructuring RFC's.
+
+
+
+## Motivation
+Aliasing table items through variable assignments is a common pattern in luau, especially when working with large libraries. However doing this with a lot of items becomes repetitve:
+```lua
+local useEffect = React.useEffect
+local useMemo = React.useMemo
+local useState = React.useState
+```
+
+With destructuring syntax this could be shortened to the following:
+```lua
+local &{ useEffect, useMemo, useState } = React
+```
+
+## Design
+
+We propose that the table destructuring syntax is prefixed with an ampersand (`&`) symbol. Without a prefix destructuring becomes ambiguous in certain scenarios, such as the one below.
```

Comment:
I do not think table destructuring should be allowed in assignments. Allowing this in assignments would introduce a footgun where you want to introduce a binding and assign a new value to some other binding.

Also, what would it even mean if you had this?
```luau
local foo

-- is `set` a no-op function?
-- or a function that captures `foo` and updates it?
function set(.{ foo }) end
```

The way to think about it is that pattern matching is generally a readonly operation, and assignments certainly isn't one. I say generally, but there are counterexamples, like [mutable matching](https://counterexamples.org/mutable-matching.html) and in the case of Luau, metatables that does side effects, but alas.